Services/ProjectServices/ProjectService.cs

```diff
@@ -12,79 +15,69 @@ public class ProjectService : IProjectService
+            if (project == null)
+                return new ServiceResponse<ProjectResponseDto> { Message = "Project not found or you are not authorized to access it", Status = ResponseStatus.NotFound };
 
             var projectResponseDto = _mapper.Map<ProjectResponseDto>(project);
-
-            return new ApiResponse<ProjectResponseDto> { Result = projectResponseDto, Message = "Project retrieved successfully", Status = ResponseStatus.Success };
+            return new ServiceResponse<ProjectResponseDto> { Result = projectResponseDto, Message = "Project retrieved successfully", Status = ResponseStatus.Success };
         }
 
-        public async Task<ApiResponse<List<ProjectResponseDto>>> GetProjectsAsync(ClaimsPrincipal userId)
+        public async Task<ServiceResponse<List<ProjectResponseDto>>> GetProjectsAsync(ClaimsPrincipal claimsPrincipal)
         {
-            var currentUser = await _userRepository.GetCurrentUserAsync(userId);
-            if (currentUser == null) return new ApiResponse<List<ProjectResponseDto>> { Message = "User not found", Status = ResponseStatus.NotFound };
+            var currentUserId = claimsPrincipal.GetLoggedInUserId();
 
-            var projects = await _projectRepository.GetProjectsAsync();
+            var projects = await _projectRepository.GetProjectsAsync(currentUserId);
             var projectResponseDtos = _mapper.Map<List<ProjectResponseDto>>(projects);
 
-            return new ApiResponse<List<ProjectResponseDto>> { Result = projectResponseDtos, Message = "Projects retrieved successfully", Status = ResponseStatus.Success };
+            return new ServiceResponse<List<ProjectResponseDto>> { Result = projectResponseDtos, Message = "Projects retrieved successfully", Status = ResponseStatus.Success };
         }
 
-        public async Task<ApiResponse<ProjectResponseDto>> UpdateProjectAsync(Guid id, ProjectRequestDto projectRequestDto, ClaimsPrincipal userId)
+        public async Task<ServiceResponse<ProjectResponseDto>> UpdateProjectAsync(Guid id, ProjectRequestDto projectRequestDto, ClaimsPrincipal claimsPrincipal)
         {
-            var currentUser = await _userRepository.GetCurrentUserAsync(userId);
-            if (currentUser == null) return new ApiResponse<ProjectResponseDto> { Message = "User not found", Status = ResponseStatus.NotFound };
-
-            var project = _mapper.Map<Project>(projectRequestDto);
-            var updatedProject = await _projectRepository.UpdateProjectAsync(id, project);
-            if (updatedProject == null) return new ApiResponse<ProjectResponseDto> { Message = "Project not found", Status = ResponseStatus.NotFound };
+            var currentUserId = claimsPrincipal.GetLoggedInUserId();
 
-            var projectResponseDto = _mapper.Map<ProjectResponseDto>(updatedProject);
+            var result = await _projectRepository.UpdateProjectAsync(id, _mapper.Map<Project>(projectRequestDto), currentUserId);
```

Comment:
would this return null if  the project exists but the user is not authorised to update it?

Reply:
It would return null if the project is not found by the Id, and the user is not authorized

Reply:
It would return null if the project is not found by the Id, and the user is not authorized